libtenzir/builtins/contexts/lookup-table.cpp

```diff
@@ -125,6 +125,28 @@ class ctx final : public virtual context {
     return record{{"num_entries", context_entries.size()}};
   }
 
+  auto dump_data_entries() const -> generator<table_slice> {
+    auto entry_builder = series_builder{};
+    for (const auto& [key, value] : context_entries) {
+      entry_builder.data(record{{"key", key}, {"value", value}});
+      auto slices = entry_builder.finish_as_table_slice();
+      for (auto&& slice : slices) {
+        co_yield slice;
+      }
+    }
```

Comment:
This does two things we shouldn't do:
- It creates batches of size 1.
- It allocates a new record, only to then transpose it back into a columnar structure when adding it to the builder.

I'd suggest rewriting the loop (1) to add key and value separately to the builder, instead of allocating an intermediate row-major representation and (2) to only finish every $`2^{16}`$ entries.